modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -636,41 +588,26 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
 			path = ibctesting.NewPath(suite.chainA, suite.chainB)
 			suite.coordinator.Setup(path)
 
-			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = mock.UpgradeVersion
-
-			err := path.EndpointB.ChanUpgradeInit()
-			suite.Require().NoError(err)
-
-			err = path.EndpointA.ChanUpgradeTry()
-			suite.Require().NoError(err)
-
-			err = path.EndpointB.ChanUpgradeAck()
-			suite.Require().NoError(err)
-
-			// TODO: Remove setting of FLUSHCOMPLETE once #3928 is completed
-			channelB := path.EndpointB.GetChannel()
-			channelB.FlushStatus = types.FLUSHCOMPLETE
-			path.EndpointB.SetChannel(channelB)
+			// create upgrade proposal with different version in init upgrade to see if the WriteUpgradeAck overwrites the version
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = "original-version"
+			proposedUpgrade = path.EndpointA.GetProposedUpgrade()
+			path.EndpointA.SetChannelUpgrade(proposedUpgrade)
```

Comment:
don't think we need to get/set here, right? Endpoint handlers should be getting it and writeupgrade init setting it

Reply:
you are correct!